#728 Encrypt user_id on account verification

OmarElgabry opened the pull request:
Instead of passing user_id in the URL as action method argument, passing the encrypted version of user_id in query parameter will work just fine. The link looks long and ugly :smile:  but in case you want a solution for #728

OmarElgabry committed:
Merge pull request #1 from panique/develop

Update Develop

OmarElgabry committed:
Encrypt user_id on account verification

OmarElgabry committed:
Remove user_id parameter

panique commented:
Thanks, looks very good!
Would be cool if everybody who reads this could test it a little bit to make sure this is bulletproof in most possible browser/mailprovider/server/os-setup (there are sometimes problems when using complex strings with special characters in URLs).

slaveek commented on application/controller/RegisterController.php:
```diff
@@ -46,11 +46,12 @@ public function register_action()
-    public function verify($user_id, $user_activation_verification_code)
+    public function verify($user_activation_verification_code)
     {
+		$user_id = Encryption::decrypt(Request::get("user_id"));
```
Request::get("user_id") should be check for empty value otherwise exception is thrown

OmarElgabry committed:
Check if user_id is empty

Check if user_id is empty before decryption, otherwise an exception will be thrown.

OmarElGabry commented:
@slaveek It's fixed now, Thanks.

borgogelli commented:
You change the sendVerificationEmail() but what about sendPasswordResetMail() ? Do I need to change it too ?

OmarElGabry commented:
The password rest uses the user name instead of user id, that's why I changed sendVerificationEmail(). You can use the user id instead in sendPasswordResetMail(), encrypt it, and don't forget to pass it as a query argument.

di48l069 commented:
Hi @panique Will you be adding this to huge?

panique commented:
Hi @di48l069, yes it's the plan but i had no time for review and merging yet :/ ... you know, working full time + summer is always tricky :)

di48l069 commented:
Hi @panique thank you for response haha I understand working full time is hectic no problem man i will just keep eye on pull request :P

Thanks for this amazing secure framework it has helped my projects alot

Kayumba commented:
v
